## Question 1:

Write a program that calculates and prints the value according to the given formula:

Q = Square root of [(2 * C * D)/H]

Following are the fixed values of C and H:

C is 50. H is 30.

D is the variable whose values should be input to your program in a comma-separated

sequence.

Example

Let us assume the following comma separated input sequence is given to the program:

100,150,180

The output of the program should be:

18,22,24


In [47]:
import math

n1 = input("enter the value of D: ")
n1 = n1.split(',')

result = []
for D in n1:
    Q = round(math.sqrt(2 * 50 * int(D) / 30))
    result.append(Q)

print(result)


enter the value of D: 100
[18]


## Question 2:

Write a program which takes 2 digits, X,Y as input and generates a 2-dimensional array. The
element value in the i-th row and j-th column of the array should be i*j.
Note: i=0,1.., X-1; j=0,1,¡Y-1.

Example

Suppose the following inputs are given to the program:

3,5

Then, the output of the program should be:

[[0, 0, 0, 0, 0], [0, 1, 2, 3, 4], [0, 2, 4, 6, 8]]

In [45]:
n2 = input("enter the value of D: ")
l=[]
for i in n2:
    l.append(i)
X = int(l[0])
Y = int(l[1])

multi_list = [[0 for col in range(Y)] for row in range(X)]

for row in range(X):
    for col in range(Y):
        multi_list[row][col]= row*col

print(multi_list)

enter the value of D: 35
[[0, 0, 0, 0, 0], [0, 1, 2, 3, 4], [0, 2, 4, 6, 8]]


## Question 3:

Write a program that accepts a comma separated sequence of words as input and prints the
words in a comma-separated sequence after sorting them alphabetically.

Suppose the following input is supplied to the program:
    
without,hello,bag,world

Then, the output should be:
    
bag,hello,without,world

In [88]:
sentence = input("insert the sentence: ")
splitsentence = sentence.split(' ')

word_list = []
for i in splitsentence:
    if i not in word_list:
        word_list.append(i)
    else:
        continue
word_list.sort()
print((' ').join(word_list))

insert the sentence: hello world and practice makes perfect and hello world again
again and hello makes perfect practice world


## Question 4:

Write a program that accepts a sequence of whitespace separated words as input and prints
the words after removing all duplicate words and sorting them alphanumerically.

Suppose the following input is supplied to the program:
    
hello world and practice makes perfect and hello world again

Then, the output should be:
    
again and hello makes perfect practice world

In [87]:
sentence = input("Insert the sentence: ")
sentence = list(sentence)

a = 0
for i in sentence:
    if i.isalpha():
        a = a + 1
    else:
        pass

print("LETTERS: ",a)

Insert the sentence: hello world! 123
LETTERS:  10


## Question 5:

Write a program that accepts a sentence and calculate the number of letters and digits.

Suppose the following input is supplied to the program:
    
hello world! 123

Then, the output should be:
    
LETTERS 10

In [86]:
#importing regular expression

import re

p = input("input list of password ")
p = p.split(",")

listpass = []
for i in p:
    
    if len(i) < 6 or len(i) > 12:
        continue

    elif not re.search("([a-z])+", i):
        continue

    elif not re.search("([A-Z])+", i):
        continue

    elif not re.search("([0-9])+", i):
        continue

    elif not re.search("([!@$%^&])+", i):
        continue

    else:
        listpass.append(i)

print((" ").join(listpass))

input list of password ABd1234@1,a F1#,2w3E*,2We3345
ABd1234@1
